{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "9d05bdce",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Collecting geoparse\n",
      "  Downloading GEOparse-2.0.3.tar.gz (278 kB)\n",
      "     ------------------------------------ 278.5/278.5 kB 264.3 kB/s eta 0:00:00\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'done'\n",
      "Requirement already satisfied: numpy>=1.7 in c:\\programdata\\anaconda3\\lib\\site-packages (from geoparse) (1.21.5)\n",
      "Requirement already satisfied: pandas>=0.17 in c:\\programdata\\anaconda3\\lib\\site-packages (from geoparse) (1.4.4)\n",
      "Requirement already satisfied: requests>=2.21.0 in c:\\programdata\\anaconda3\\lib\\site-packages (from geoparse) (2.28.1)\n",
      "Requirement already satisfied: tqdm>=4.31.1 in c:\\programdata\\anaconda3\\lib\\site-packages (from geoparse) (4.64.1)\n",
      "Requirement already satisfied: pytz>=2020.1 in c:\\programdata\\anaconda3\\lib\\site-packages (from pandas>=0.17->geoparse) (2022.7)\n",
      "Requirement already satisfied: python-dateutil>=2.8.1 in c:\\programdata\\anaconda3\\lib\\site-packages (from pandas>=0.17->geoparse) (2.8.2)\n",
      "Requirement already satisfied: idna<4,>=2.5 in c:\\programdata\\anaconda3\\lib\\site-packages (from requests>=2.21.0->geoparse) (3.4)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in c:\\programdata\\anaconda3\\lib\\site-packages (from requests>=2.21.0->geoparse) (2022.12.7)\n",
      "Requirement already satisfied: charset-normalizer<3,>=2 in c:\\programdata\\anaconda3\\lib\\site-packages (from requests>=2.21.0->geoparse) (2.0.4)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in c:\\programdata\\anaconda3\\lib\\site-packages (from requests>=2.21.0->geoparse) (1.26.13)\n",
      "Requirement already satisfied: colorama in c:\\programdata\\anaconda3\\lib\\site-packages (from tqdm>=4.31.1->geoparse) (0.4.6)\n",
      "Requirement already satisfied: six>=1.5 in c:\\programdata\\anaconda3\\lib\\site-packages (from python-dateutil>=2.8.1->pandas>=0.17->geoparse) (1.16.0)\n",
      "Building wheels for collected packages: geoparse\n",
      "  Building wheel for geoparse (setup.py): started\n",
      "  Building wheel for geoparse (setup.py): finished with status 'done'\n",
      "  Created wheel for geoparse: filename=GEOparse-2.0.3-py3-none-any.whl size=29065 sha256=7fe408b62179bea5b3f4078fe6f4db271df8520b09ab0e4115b41bb65d07e24d\n",
      "  Stored in directory: c:\\users\\friends.desktop-1rirfcp\\appdata\\local\\pip\\cache\\wheels\\1a\\1b\\e7\\d81777c58681ce6d311a00f381565c2d43032c43b6dd34e596\n",
      "Successfully built geoparse\n",
      "Installing collected packages: geoparse\n",
      "Successfully installed geoparse-2.0.3\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip install geoparse\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1bd19835",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "01-May-2023 15:59:59 INFO GEOparse - Downloading ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE32nnn/GSE32861/soft/GSE32861_family.soft.gz to /bin/LungCancer\\GSE32861_family.soft.gz\n",
      "100%|██████████| 36.1M/36.1M [00:09<00:00, 3.83MB/s]  \n",
      "01-May-2023 16:00:10 DEBUG downloader - Size validation passed\n",
      "01-May-2023 16:00:10 DEBUG downloader - Moving C:\\Users\\FRIEND~1.DES\\AppData\\Local\\Temp\\tmprd1ijxc9 to C:\\bin\\LungCancer\\GSE32861_family.soft.gz\n",
      "01-May-2023 16:00:11 DEBUG downloader - Successfully downloaded ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE32nnn/GSE32861/soft/GSE32861_family.soft.gz\n",
      "01-May-2023 16:00:11 INFO GEOparse - Parsing /bin/LungCancer\\GSE32861_family.soft.gz: \n",
      "01-May-2023 16:00:11 DEBUG GEOparse - DATABASE: GeoMiame\n",
      "01-May-2023 16:00:11 DEBUG GEOparse - SERIES: GSE32861\n",
      "01-May-2023 16:00:11 DEBUG GEOparse - PLATFORM: GPL8490\n",
      "01-May-2023 16:00:12 DEBUG GEOparse - SAMPLE: GSM813174\n",
      "01-May-2023 16:00:12 DEBUG GEOparse - SAMPLE: GSM813175\n",
      "01-May-2023 16:00:12 DEBUG GEOparse - SAMPLE: GSM813176\n",
      "01-May-2023 16:00:12 DEBUG GEOparse - SAMPLE: GSM813177\n",
      "01-May-2023 16:00:13 DEBUG GEOparse - SAMPLE: GSM813178\n",
      "01-May-2023 16:00:13 DEBUG GEOparse - SAMPLE: GSM813179\n",
      "01-May-2023 16:00:13 DEBUG GEOparse - SAMPLE: GSM813180\n",
      "01-May-2023 16:00:13 DEBUG GEOparse - SAMPLE: GSM813181\n",
      "01-May-2023 16:00:13 DEBUG GEOparse - SAMPLE: GSM813182\n",
      "01-May-2023 16:00:13 DEBUG GEOparse - SAMPLE: GSM813183\n",
      "01-May-2023 16:00:14 DEBUG GEOparse - SAMPLE: GSM813184\n",
      "01-May-2023 16:00:14 DEBUG GEOparse - SAMPLE: GSM813185\n",
      "01-May-2023 16:00:14 DEBUG GEOparse - SAMPLE: GSM813186\n",
      "01-May-2023 16:00:14 DEBUG GEOparse - SAMPLE: GSM813187\n",
      "01-May-2023 16:00:14 DEBUG GEOparse - SAMPLE: GSM813188\n",
      "01-May-2023 16:00:14 DEBUG GEOparse - SAMPLE: GSM813189\n",
      "01-May-2023 16:00:14 DEBUG GEOparse - SAMPLE: GSM813190\n",
      "01-May-2023 16:00:14 DEBUG GEOparse - SAMPLE: GSM813191\n",
      "01-May-2023 16:00:15 DEBUG GEOparse - SAMPLE: GSM813192\n",
      "01-May-2023 16:00:15 DEBUG GEOparse - SAMPLE: GSM813193\n",
      "01-May-2023 16:00:15 DEBUG GEOparse - SAMPLE: GSM813194\n",
      "01-May-2023 16:00:15 DEBUG GEOparse - SAMPLE: GSM813195\n",
      "01-May-2023 16:00:15 DEBUG GEOparse - SAMPLE: GSM813196\n",
      "01-May-2023 16:00:15 DEBUG GEOparse - SAMPLE: GSM813197\n",
      "01-May-2023 16:00:15 DEBUG GEOparse - SAMPLE: GSM813198\n",
      "01-May-2023 16:00:15 DEBUG GEOparse - SAMPLE: GSM813199\n",
      "01-May-2023 16:00:16 DEBUG GEOparse - SAMPLE: GSM813200\n",
      "01-May-2023 16:00:16 DEBUG GEOparse - SAMPLE: GSM813201\n",
      "01-May-2023 16:00:16 DEBUG GEOparse - SAMPLE: GSM813202\n",
      "01-May-2023 16:00:16 DEBUG GEOparse - SAMPLE: GSM813203\n",
      "01-May-2023 16:00:16 DEBUG GEOparse - SAMPLE: GSM813204\n",
      "01-May-2023 16:00:16 DEBUG GEOparse - SAMPLE: GSM813205\n",
      "01-May-2023 16:00:17 DEBUG GEOparse - SAMPLE: GSM813206\n",
      "01-May-2023 16:00:17 DEBUG GEOparse - SAMPLE: GSM813207\n",
      "01-May-2023 16:00:17 DEBUG GEOparse - SAMPLE: GSM813208\n",
      "01-May-2023 16:00:17 DEBUG GEOparse - SAMPLE: GSM813209\n",
      "01-May-2023 16:00:17 DEBUG GEOparse - SAMPLE: GSM813210\n",
      "01-May-2023 16:00:17 DEBUG GEOparse - SAMPLE: GSM813211\n",
      "01-May-2023 16:00:17 DEBUG GEOparse - SAMPLE: GSM813212\n",
      "01-May-2023 16:00:18 DEBUG GEOparse - SAMPLE: GSM813213\n",
      "01-May-2023 16:00:18 DEBUG GEOparse - SAMPLE: GSM813214\n",
      "01-May-2023 16:00:18 DEBUG GEOparse - SAMPLE: GSM813215\n",
      "01-May-2023 16:00:18 DEBUG GEOparse - SAMPLE: GSM813216\n",
      "01-May-2023 16:00:18 DEBUG GEOparse - SAMPLE: GSM813217\n",
      "01-May-2023 16:00:18 DEBUG GEOparse - SAMPLE: GSM813218\n",
      "01-May-2023 16:00:18 DEBUG GEOparse - SAMPLE: GSM813219\n",
      "01-May-2023 16:00:19 DEBUG GEOparse - SAMPLE: GSM813220\n",
      "01-May-2023 16:00:19 DEBUG GEOparse - SAMPLE: GSM813221\n",
      "01-May-2023 16:00:19 DEBUG GEOparse - SAMPLE: GSM813222\n",
      "01-May-2023 16:00:19 DEBUG GEOparse - SAMPLE: GSM813223\n",
      "01-May-2023 16:00:19 DEBUG GEOparse - SAMPLE: GSM813224\n",
      "01-May-2023 16:00:19 DEBUG GEOparse - SAMPLE: GSM813225\n",
      "01-May-2023 16:00:19 DEBUG GEOparse - SAMPLE: GSM813226\n",
      "01-May-2023 16:00:19 DEBUG GEOparse - SAMPLE: GSM813227\n",
      "01-May-2023 16:00:19 DEBUG GEOparse - SAMPLE: GSM813228\n",
      "01-May-2023 16:00:20 DEBUG GEOparse - SAMPLE: GSM813229\n",
      "01-May-2023 16:00:20 DEBUG GEOparse - SAMPLE: GSM813230\n",
      "01-May-2023 16:00:20 DEBUG GEOparse - SAMPLE: GSM813231\n",
      "01-May-2023 16:00:20 DEBUG GEOparse - SAMPLE: GSM813232\n",
      "01-May-2023 16:00:20 DEBUG GEOparse - SAMPLE: GSM813233\n",
      "01-May-2023 16:00:20 DEBUG GEOparse - SAMPLE: GSM813234\n",
      "01-May-2023 16:00:20 DEBUG GEOparse - SAMPLE: GSM813235\n",
      "01-May-2023 16:00:20 DEBUG GEOparse - SAMPLE: GSM813236\n",
      "01-May-2023 16:00:21 DEBUG GEOparse - SAMPLE: GSM813237\n",
      "01-May-2023 16:00:21 DEBUG GEOparse - SAMPLE: GSM813238\n",
      "01-May-2023 16:00:21 DEBUG GEOparse - SAMPLE: GSM813239\n",
      "01-May-2023 16:00:21 DEBUG GEOparse - SAMPLE: GSM813240\n",
      "01-May-2023 16:00:21 DEBUG GEOparse - SAMPLE: GSM813241\n",
      "01-May-2023 16:00:21 DEBUG GEOparse - SAMPLE: GSM813242\n",
      "01-May-2023 16:00:21 DEBUG GEOparse - SAMPLE: GSM813243\n",
      "01-May-2023 16:00:21 DEBUG GEOparse - SAMPLE: GSM813244\n",
      "01-May-2023 16:00:21 DEBUG GEOparse - SAMPLE: GSM813245\n",
      "01-May-2023 16:00:22 DEBUG GEOparse - SAMPLE: GSM813246\n",
      "01-May-2023 16:00:22 DEBUG GEOparse - SAMPLE: GSM813247\n",
      "01-May-2023 16:00:22 DEBUG GEOparse - SAMPLE: GSM813248\n",
      "01-May-2023 16:00:22 DEBUG GEOparse - SAMPLE: GSM813249\n",
      "01-May-2023 16:00:22 DEBUG GEOparse - SAMPLE: GSM813250\n",
      "01-May-2023 16:00:22 DEBUG GEOparse - SAMPLE: GSM813251\n",
      "01-May-2023 16:00:22 DEBUG GEOparse - SAMPLE: GSM813252\n",
      "01-May-2023 16:00:23 DEBUG GEOparse - SAMPLE: GSM813253\n",
      "01-May-2023 16:00:23 DEBUG GEOparse - SAMPLE: GSM813254\n",
      "01-May-2023 16:00:23 DEBUG GEOparse - SAMPLE: GSM813255\n",
      "01-May-2023 16:00:23 DEBUG GEOparse - SAMPLE: GSM813256\n",
      "01-May-2023 16:00:23 DEBUG GEOparse - SAMPLE: GSM813257\n",
      "01-May-2023 16:00:23 DEBUG GEOparse - SAMPLE: GSM813258\n",
      "01-May-2023 16:00:23 DEBUG GEOparse - SAMPLE: GSM813259\n",
      "01-May-2023 16:00:24 DEBUG GEOparse - SAMPLE: GSM813260\n",
      "01-May-2023 16:00:24 DEBUG GEOparse - SAMPLE: GSM813261\n",
      "01-May-2023 16:00:24 DEBUG GEOparse - SAMPLE: GSM813262\n",
      "01-May-2023 16:00:24 DEBUG GEOparse - SAMPLE: GSM813263\n",
      "01-May-2023 16:00:24 DEBUG GEOparse - SAMPLE: GSM813264\n",
      "01-May-2023 16:00:24 DEBUG GEOparse - SAMPLE: GSM813265\n",
      "01-May-2023 16:00:24 DEBUG GEOparse - SAMPLE: GSM813266\n",
      "01-May-2023 16:00:24 DEBUG GEOparse - SAMPLE: GSM813267\n",
      "01-May-2023 16:00:25 DEBUG GEOparse - SAMPLE: GSM813268\n",
      "01-May-2023 16:00:25 DEBUG GEOparse - SAMPLE: GSM813269\n",
      "01-May-2023 16:00:25 DEBUG GEOparse - SAMPLE: GSM813270\n",
      "01-May-2023 16:00:25 DEBUG GEOparse - SAMPLE: GSM813271\n",
      "01-May-2023 16:00:25 DEBUG GEOparse - SAMPLE: GSM813272\n",
      "01-May-2023 16:00:25 DEBUG GEOparse - SAMPLE: GSM813273\n",
      "01-May-2023 16:00:25 DEBUG GEOparse - SAMPLE: GSM813274\n",
      "01-May-2023 16:00:25 DEBUG GEOparse - SAMPLE: GSM813275\n",
      "01-May-2023 16:00:26 DEBUG GEOparse - SAMPLE: GSM813276\n",
      "01-May-2023 16:00:26 DEBUG GEOparse - SAMPLE: GSM813277\n",
      "01-May-2023 16:00:26 DEBUG GEOparse - SAMPLE: GSM813278\n",
      "01-May-2023 16:00:26 DEBUG GEOparse - SAMPLE: GSM813279\n",
      "01-May-2023 16:00:26 DEBUG GEOparse - SAMPLE: GSM813280\n",
      "01-May-2023 16:00:26 DEBUG GEOparse - SAMPLE: GSM813281\n",
      "01-May-2023 16:00:27 DEBUG GEOparse - SAMPLE: GSM813282\n",
      "01-May-2023 16:00:27 DEBUG GEOparse - SAMPLE: GSM813283\n",
      "01-May-2023 16:00:27 DEBUG GEOparse - SAMPLE: GSM813284\n",
      "01-May-2023 16:00:27 DEBUG GEOparse - SAMPLE: GSM813285\n",
      "01-May-2023 16:00:27 DEBUG GEOparse - SAMPLE: GSM813286\n",
      "01-May-2023 16:00:28 DEBUG GEOparse - SAMPLE: GSM813287\n",
      "01-May-2023 16:00:28 DEBUG GEOparse - SAMPLE: GSM813288\n",
      "01-May-2023 16:00:28 DEBUG GEOparse - SAMPLE: GSM813289\n",
      "01-May-2023 16:00:28 DEBUG GEOparse - SAMPLE: GSM813290\n",
      "01-May-2023 16:00:28 DEBUG GEOparse - SAMPLE: GSM813291\n"
     ]
    }
   ],
   "source": [
    "import GEOparse\n",
    "\n",
    "gse = GEOparse.get_GEO(geo=\"GSE32861\", destdir=\"/bin/LungCancer\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "03af69a2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "GSM example:\n",
      "Name:  GSM813174\n",
      "Metadata:\n",
      " - title : Adjacent non-tumor lung 05L12_N (methylation)\n",
      " - geo_accession : GSM813174\n",
      " - status : Public on Mar 21 2012\n",
      " - submission_date : Oct 11 2011\n",
      " - last_update_date : Mar 21 2012\n",
      " - type : genomic\n",
      " - channel_count : 1\n",
      " - source_name_ch1 : Adjacent non-tumor\n",
      " - organism_ch1 : Homo sapiens\n",
      " - taxid_ch1 : 9606\n",
      " - characteristics_ch1 : sample id: N7, tissue source: nci-edrn, bisulfite plate: 32, beadchip: 8, beadstrip: E, bisulfite alu: 12.16426, age: 66, tissue: Normal lung, gender: Male, ethnicity: Asian, pack years: 0, smoking status: Never, Stage: Stage IIIA, recurrence: Yes, subject: 05L12\n",
      " - treatment_protocol_ch1 : Bisulfite treatment\n",
      " - growth_protocol_ch1 : Fresh-frozen tissue\n",
      " - molecule_ch1 : genomic DNA\n",
      " - extract_protocol_ch1 : Genomic DNA was extracted after macrodissection and a proteinase K digstion by  phenol-chloroform\n",
      " - label_ch1 : Cy3 and Cy5\n",
      " - label_protocol_ch1 : Standard Illumina protocol\n",
      " - hyb_protocol : Standard Illumina protocol performed at the USC Epigenome Center\n",
      " - scan_protocol : Standard Illumina scan protocol performed at the USC Epigenome Center\n",
      " - data_processing : DNA methylation levels are reported as  Î²-values, calculated from mean non-background corrected methylated (M) and unmethylated (U) signal intensities for each locus for each sample using the formula (Î²=M/(M+U)) and the Illumina BeadStudio software v3.2. Probes with detection p-values of >0.05 as described previously (Noushemehr et al., 2010) were deemed not significantly different from background noise and were labeled â€œNAâ€.\n",
      " - platform_id : GPL8490\n",
      " - contact_name : Ite,,Laird-Offringa\n",
      " - contact_email : ilaird@usc.edu\n",
      " - contact_phone : (323) 865-0655\n",
      " - contact_fax : (323) 865-0158\n",
      " - contact_laboratory : Ite Laird-Offringa\n",
      " - contact_department : Biochemistry\n",
      " - contact_institute : University of Southern California\n",
      " - contact_address : 1441 Eastlake Avenue, Suite 6420\n",
      " - contact_city : Los Angeles\n",
      " - contact_state : CA\n",
      " - contact_zip/postal_code : 90089\n",
      " - contact_country : USA\n",
      " - supplementary_file : NONE\n",
      " - series_id : GSE32861, GSE32867\n",
      " - data_row_count : 27578\n",
      "Table data:\n",
      "       ID_REF     VALUE  Detection Pval\n",
      "0  cg00000292  0.619946             0.0\n",
      "1  cg00002426  0.460933             0.0\n",
      "2  cg00003994  0.051960             0.0\n",
      "3  cg00005847  0.240433             0.0\n",
      "4  cg00006414  0.013831             0.0\n",
      "\n",
      "GPL example:\n",
      "Name:  GPL8490\n",
      "Metadata:\n",
      " - title : Illumina HumanMethylation27 BeadChip (HumanMethylation27_270596_v.1.2)\n",
      " - geo_accession : GPL8490\n",
      " - status : Public on Apr 27 2009\n",
      " - submission_date : Apr 27 2009\n",
      " - last_update_date : Jan 02 2015\n",
      " - technology : oligonucleotide beads\n",
      " - distribution : commercial\n",
      " - organism : Homo sapiens\n",
      " - taxid : 9606\n",
      " - manufacturer : Illumina, Inc.\n",
      " - manufacture_protocol : See manufacturer's website\n",
      " - description : HumanMethylation27 DNA Analysis BeadChip allows researchers to interrogate 27,578 highly informative CpG sites per sample at single-nucleotide resolution. This 12-sample BeadChip features content derived from the well-annotated NCBI CCDS database (Genome Build 36) and is supplemented with more than 1,000 cancer-related genes described in published literature. Probe content has been enriched to deeply cover more than 150 well-established cancer genes known to show differential methylation patterns. HumanMethylation27 BeadChip content also targets the promoter regions of 110 miRNA genes.\n",
      " - web_link : http://www.illumina.com/pages.ilmn?ID=243\n",
      " - contact_name : ,,Illumina Inc.\n",
      " - contact_email : expression@illumina.com, techsupport@illumina.com\n",
      " - contact_phone : 1 800 809 4566\n",
      " - contact_institute : Illumina Inc.\n",
      " - contact_address : 9885 Towne Centre Drive\n",
      " - contact_city : San Diego\n",
      " - contact_state : CA\n",
      " - contact_zip/postal_code : 92121\n",
      " - contact_country : USA\n",
      " - contact_web_link : www.illumina.com\n",
      " - supplementary_file : ftp://ftp.ncbi.nlm.nih.gov/geo/platforms/GPL8nnn/GPL8490/suppl/GPL8490_HumanMethylation27_270596_v.1.2.csv.gz\n",
      " - data_row_count : 27578\n",
      "Table data:\n",
      "           ID        Name IlmnStrand  AddressA_ID  \\\n",
      "0  cg00000292  cg00000292        TOP       990370   \n",
      "1  cg00002426  cg00002426        TOP      6580397   \n",
      "2  cg00003994  cg00003994        TOP      7150184   \n",
      "3  cg00005847  cg00005847        BOT      4850717   \n",
      "4  cg00006414  cg00006414        BOT      6980731   \n",
      "\n",
      "                                    AlleleA_ProbeSeq  AddressB_ID  \\\n",
      "0  AAACATTAATTACCAACCACTCTTCCAAAAAACACTTACCATTAAA...      6660678   \n",
      "1  AATATAATAACATTACCTTACCCATCTTATAATCAAACCAAACAAA...      6100343   \n",
      "2  AATAATAATAATACCCCCTATAATACTAACTAACAAACATACCCTC...      7150392   \n",
      "3  TACTATAATACACCCTATATTTAAAACACTAAACTTACCCCATTAA...      1260113   \n",
      "4  CTCAAAAACCAAACAAAACAAAACCCCAATACTAATCATTAATAAA...      4280093   \n",
      "\n",
      "                                    AlleleB_ProbeSeq  GenomeBuild Chr  \\\n",
      "0  AAACATTAATTACCAACCGCTCTTCCAAAAAACACTTACCATTAAA...           36  16   \n",
      "1  AATATAATAACATTACCTTACCCGTCTTATAATCAAACCAAACGAA...           36   3   \n",
      "2  AATAATAATAATACCCCCTATAATACTAACTAACAAACATACCCTC...           36   7   \n",
      "3  TACTATAATACACCCTATATTTAAAACACTAAACTTACCCCATTAA...           36   2   \n",
      "4  CTCGAAAACCGAACAAAACAAAACCCCAATACTAATCGTTAATAAA...           36   7   \n",
      "\n",
      "     MapInfo  ... Distance_to_TSS CPG_ISLAND   CPG_ISLAND_LOCATIONS  \\\n",
      "0   28797601  ...           291.0       True   16:28797486-28797825   \n",
      "1   57718583  ...           369.0       True    3:57716811-57718675   \n",
      "2   15692387  ...           432.0       True    7:15691512-15693551   \n",
      "3  176737319  ...           268.0      False                    NaN   \n",
      "4  148453770  ...           671.0       True  7:148453584-148455804   \n",
      "\n",
      "   MIR_CPG_ISLAND      RANGE_GB  RANGE_START    RANGE_END RANGE_STRAND  \\\n",
      "0             NaN   NC_000016.8   28797486.0   28797825.0            +   \n",
      "1             NaN  NC_000003.10   57716811.0   57718675.0            +   \n",
      "2             NaN  NC_000007.12   15691512.0   15693551.0            -   \n",
      "3             NaN           NaN          NaN          NaN          NaN   \n",
      "4             NaN  NC_000007.12  148453584.0  148455804.0            +   \n",
      "\n",
      "        GB_ACC    ORF  \n",
      "0  NM_173201.2    487  \n",
      "1  NM_007159.2   7871  \n",
      "2  NM_005924.3   4223  \n",
      "3  NM_006898.4   3232  \n",
      "4  NM_020781.2  57541  \n",
      "\n",
      "[5 rows x 38 columns]\n"
     ]
    }
   ],
   "source": [
    "print()\n",
    "print(\"GSM example:\")\n",
    "for gsm_name, gsm in gse.gsms.items():\n",
    "    print(\"Name: \", gsm_name)\n",
    "    print(\"Metadata:\",)\n",
    "    for key, value in gsm.metadata.items():\n",
    "        print(\" - %s : %s\" % (key, \", \".join(value)))\n",
    "    print (\"Table data:\",)\n",
    "    print (gsm.table.head())\n",
    "    break\n",
    "\n",
    "print()\n",
    "print(\"GPL example:\")\n",
    "for gpl_name, gpl in gse.gpls.items():\n",
    "    print(\"Name: \", gpl_name)\n",
    "    print(\"Metadata:\",)\n",
    "    for key, value in gpl.metadata.items():\n",
    "        print(\" - %s : %s\" % (key, \", \".join(value)))\n",
    "    print(\"Table data:\",)\n",
    "    print(gpl.table.head())\n",
    "    break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "968b9388",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "tissue = gse.phenotype_data['characteristics_ch1.7.tissue']\n",
    "tissue = tissue.rename(\"tissue_char\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "b0e9fba0",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\FRIENDS.DESKTOP-1RIRFCP\\AppData\\Local\\Temp\\ipykernel_7160\\1717902988.py:2: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  data_matrix = data_matrix.append(tissue)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID_REF    cg00000292 cg00002426 cg00003994 cg00005847 cg00006414 cg00007981  \\\n",
      "GSM813174   0.619946   0.460933    0.05196   0.240433   0.013831   0.023905   \n",
      "GSM813175    0.42915   0.361039   0.066462   0.712471   0.105384   0.146959   \n",
      "GSM813176   0.598473   0.399592   0.081039    0.19372   0.039534   0.024417   \n",
      "GSM813177   0.605416   0.596668   0.137904   0.241135   0.035636   0.015306   \n",
      "GSM813178   0.587209   0.436555   0.049907   0.179162   0.017098    0.02515   \n",
      "...              ...        ...        ...        ...        ...        ...   \n",
      "GSM813287    0.58477   0.608247    0.17715   0.386829   0.024344    0.03662   \n",
      "GSM813288   0.637604   0.413775    0.06929   0.211992    0.01932   0.026035   \n",
      "GSM813289   0.624775   0.689305   0.305079   0.516924   0.150818     0.1096   \n",
      "GSM813290   0.773797   0.526281    0.07592   0.233635   0.046912    0.04386   \n",
      "GSM813291     0.4425   0.367014   0.109204   0.503786   0.028654   0.034655   \n",
      "\n",
      "ID_REF    cg00008493 cg00008713 cg00009407 cg00010193  ... cg27654142  \\\n",
      "GSM813174   0.990165   0.014497   0.006952   0.662304  ...   0.093868   \n",
      "GSM813175   0.988713   0.026515   0.031751   0.588816  ...   0.054866   \n",
      "GSM813176   0.984537   0.020919   0.013639   0.663489  ...   0.053182   \n",
      "GSM813177   0.988111   0.027033   0.008842   0.675272  ...   0.097044   \n",
      "GSM813178   0.990127   0.007726   0.008732   0.603307  ...   0.076322   \n",
      "...              ...        ...        ...        ...  ...        ...   \n",
      "GSM813287   0.976659   0.010312   0.007879    0.64371  ...   0.032158   \n",
      "GSM813288   0.986941   0.012445   0.011437   0.680988  ...    0.03265   \n",
      "GSM813289   0.952523   0.037228   0.043694   0.660779  ...   0.165378   \n",
      "GSM813290   0.982608   0.022921   0.021091   0.709384  ...   0.154565   \n",
      "GSM813291   0.807651   0.023715   0.018089   0.653145  ...    0.10791   \n",
      "\n",
      "ID_REF    cg27655855 cg27655905 cg27657249 cg27657283 cg27661264 cg27662379  \\\n",
      "GSM813174   0.816186   0.033004   0.170945   0.034743   0.370746   0.006512   \n",
      "GSM813175   0.148381   0.083458   0.102413   0.137931   0.339054   0.041294   \n",
      "GSM813176   0.800741   0.045924   0.137161   0.052218   0.396524   0.012065   \n",
      "GSM813177   0.812453   0.050538   0.174674   0.031456   0.292604   0.009658   \n",
      "GSM813178   0.828708   0.031439   0.202882   0.035126   0.261081   0.009629   \n",
      "...              ...        ...        ...        ...        ...        ...   \n",
      "GSM813287   0.677973   0.031384   0.156193   0.054776   0.395179   0.013456   \n",
      "GSM813288   0.772188   0.034177    0.18395   0.048791   0.301718    0.00785   \n",
      "GSM813289   0.763463   0.124898   0.179512    0.15737   0.444144    0.06769   \n",
      "GSM813290   0.852972   0.077501   0.156859    0.03269   0.387466   0.019694   \n",
      "GSM813291   0.450145   0.056143   0.142126   0.055615      0.493   0.009382   \n",
      "\n",
      "ID_REF    cg27662877 cg27665659  tissue_char  \n",
      "GSM813174     0.0124   0.066593  Normal lung  \n",
      "GSM813175   0.199616    0.03218   Lung tumor  \n",
      "GSM813176   0.023331   0.059368  Normal lung  \n",
      "GSM813177   0.026601   0.037425   Lung tumor  \n",
      "GSM813178   0.013818   0.060106  Normal lung  \n",
      "...              ...        ...          ...  \n",
      "GSM813287   0.035263   0.054444   Lung tumor  \n",
      "GSM813288   0.017347   0.068875  Normal lung  \n",
      "GSM813289   0.107839   0.063127   Lung tumor  \n",
      "GSM813290   0.047126   0.046525  Normal lung  \n",
      "GSM813291   0.039019   0.033644   Lung tumor  \n",
      "\n",
      "[118 rows x 26301 columns]\n"
     ]
    }
   ],
   "source": [
    "data_matrix = gse.pivot_samples('VALUE')\n",
    "data_matrix = data_matrix.append(tissue)\n",
    "data_matrix_nf = data_matrix.dropna()\n",
    "data_matrix_nf = data_matrix_nf.T\n",
    "print(data_matrix_nf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "d992a527",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>ID_REF</th>\n",
       "      <th>cg00000292</th>\n",
       "      <th>cg00002426</th>\n",
       "      <th>cg00003994</th>\n",
       "      <th>cg00005847</th>\n",
       "      <th>cg00006414</th>\n",
       "      <th>cg00007981</th>\n",
       "      <th>cg00008493</th>\n",
       "      <th>cg00008713</th>\n",
       "      <th>cg00009407</th>\n",
       "      <th>cg00010193</th>\n",
       "      <th>...</th>\n",
       "      <th>cg27653134</th>\n",
       "      <th>cg27654142</th>\n",
       "      <th>cg27655855</th>\n",
       "      <th>cg27655905</th>\n",
       "      <th>cg27657249</th>\n",
       "      <th>cg27657283</th>\n",
       "      <th>cg27661264</th>\n",
       "      <th>cg27662379</th>\n",
       "      <th>cg27662877</th>\n",
       "      <th>cg27665659</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GSM813174</th>\n",
       "      <td>0.619946</td>\n",
       "      <td>0.460933</td>\n",
       "      <td>0.05196</td>\n",
       "      <td>0.240433</td>\n",
       "      <td>0.013831</td>\n",
       "      <td>0.023905</td>\n",
       "      <td>0.990165</td>\n",
       "      <td>0.014497</td>\n",
       "      <td>0.006952</td>\n",
       "      <td>0.662304</td>\n",
       "      <td>...</td>\n",
       "      <td>0.695425</td>\n",
       "      <td>0.093868</td>\n",
       "      <td>0.816186</td>\n",
       "      <td>0.033004</td>\n",
       "      <td>0.170945</td>\n",
       "      <td>0.034743</td>\n",
       "      <td>0.370746</td>\n",
       "      <td>0.006512</td>\n",
       "      <td>0.0124</td>\n",
       "      <td>0.066593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM813175</th>\n",
       "      <td>0.42915</td>\n",
       "      <td>0.361039</td>\n",
       "      <td>0.066462</td>\n",
       "      <td>0.712471</td>\n",
       "      <td>0.105384</td>\n",
       "      <td>0.146959</td>\n",
       "      <td>0.988713</td>\n",
       "      <td>0.026515</td>\n",
       "      <td>0.031751</td>\n",
       "      <td>0.588816</td>\n",
       "      <td>...</td>\n",
       "      <td>0.619898</td>\n",
       "      <td>0.054866</td>\n",
       "      <td>0.148381</td>\n",
       "      <td>0.083458</td>\n",
       "      <td>0.102413</td>\n",
       "      <td>0.137931</td>\n",
       "      <td>0.339054</td>\n",
       "      <td>0.041294</td>\n",
       "      <td>0.199616</td>\n",
       "      <td>0.03218</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM813176</th>\n",
       "      <td>0.598473</td>\n",
       "      <td>0.399592</td>\n",
       "      <td>0.081039</td>\n",
       "      <td>0.19372</td>\n",
       "      <td>0.039534</td>\n",
       "      <td>0.024417</td>\n",
       "      <td>0.984537</td>\n",
       "      <td>0.020919</td>\n",
       "      <td>0.013639</td>\n",
       "      <td>0.663489</td>\n",
       "      <td>...</td>\n",
       "      <td>0.826355</td>\n",
       "      <td>0.053182</td>\n",
       "      <td>0.800741</td>\n",
       "      <td>0.045924</td>\n",
       "      <td>0.137161</td>\n",
       "      <td>0.052218</td>\n",
       "      <td>0.396524</td>\n",
       "      <td>0.012065</td>\n",
       "      <td>0.023331</td>\n",
       "      <td>0.059368</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM813177</th>\n",
       "      <td>0.605416</td>\n",
       "      <td>0.596668</td>\n",
       "      <td>0.137904</td>\n",
       "      <td>0.241135</td>\n",
       "      <td>0.035636</td>\n",
       "      <td>0.015306</td>\n",
       "      <td>0.988111</td>\n",
       "      <td>0.027033</td>\n",
       "      <td>0.008842</td>\n",
       "      <td>0.675272</td>\n",
       "      <td>...</td>\n",
       "      <td>0.796165</td>\n",
       "      <td>0.097044</td>\n",
       "      <td>0.812453</td>\n",
       "      <td>0.050538</td>\n",
       "      <td>0.174674</td>\n",
       "      <td>0.031456</td>\n",
       "      <td>0.292604</td>\n",
       "      <td>0.009658</td>\n",
       "      <td>0.026601</td>\n",
       "      <td>0.037425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM813178</th>\n",
       "      <td>0.587209</td>\n",
       "      <td>0.436555</td>\n",
       "      <td>0.049907</td>\n",
       "      <td>0.179162</td>\n",
       "      <td>0.017098</td>\n",
       "      <td>0.02515</td>\n",
       "      <td>0.990127</td>\n",
       "      <td>0.007726</td>\n",
       "      <td>0.008732</td>\n",
       "      <td>0.603307</td>\n",
       "      <td>...</td>\n",
       "      <td>0.704033</td>\n",
       "      <td>0.076322</td>\n",
       "      <td>0.828708</td>\n",
       "      <td>0.031439</td>\n",
       "      <td>0.202882</td>\n",
       "      <td>0.035126</td>\n",
       "      <td>0.261081</td>\n",
       "      <td>0.009629</td>\n",
       "      <td>0.013818</td>\n",
       "      <td>0.060106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM813287</th>\n",
       "      <td>0.58477</td>\n",
       "      <td>0.608247</td>\n",
       "      <td>0.17715</td>\n",
       "      <td>0.386829</td>\n",
       "      <td>0.024344</td>\n",
       "      <td>0.03662</td>\n",
       "      <td>0.976659</td>\n",
       "      <td>0.010312</td>\n",
       "      <td>0.007879</td>\n",
       "      <td>0.64371</td>\n",
       "      <td>...</td>\n",
       "      <td>0.458203</td>\n",
       "      <td>0.032158</td>\n",
       "      <td>0.677973</td>\n",
       "      <td>0.031384</td>\n",
       "      <td>0.156193</td>\n",
       "      <td>0.054776</td>\n",
       "      <td>0.395179</td>\n",
       "      <td>0.013456</td>\n",
       "      <td>0.035263</td>\n",
       "      <td>0.054444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM813288</th>\n",
       "      <td>0.637604</td>\n",
       "      <td>0.413775</td>\n",
       "      <td>0.06929</td>\n",
       "      <td>0.211992</td>\n",
       "      <td>0.01932</td>\n",
       "      <td>0.026035</td>\n",
       "      <td>0.986941</td>\n",
       "      <td>0.012445</td>\n",
       "      <td>0.011437</td>\n",
       "      <td>0.680988</td>\n",
       "      <td>...</td>\n",
       "      <td>0.693108</td>\n",
       "      <td>0.03265</td>\n",
       "      <td>0.772188</td>\n",
       "      <td>0.034177</td>\n",
       "      <td>0.18395</td>\n",
       "      <td>0.048791</td>\n",
       "      <td>0.301718</td>\n",
       "      <td>0.00785</td>\n",
       "      <td>0.017347</td>\n",
       "      <td>0.068875</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM813289</th>\n",
       "      <td>0.624775</td>\n",
       "      <td>0.689305</td>\n",
       "      <td>0.305079</td>\n",
       "      <td>0.516924</td>\n",
       "      <td>0.150818</td>\n",
       "      <td>0.1096</td>\n",
       "      <td>0.952523</td>\n",
       "      <td>0.037228</td>\n",
       "      <td>0.043694</td>\n",
       "      <td>0.660779</td>\n",
       "      <td>...</td>\n",
       "      <td>0.582403</td>\n",
       "      <td>0.165378</td>\n",
       "      <td>0.763463</td>\n",
       "      <td>0.124898</td>\n",
       "      <td>0.179512</td>\n",
       "      <td>0.15737</td>\n",
       "      <td>0.444144</td>\n",
       "      <td>0.06769</td>\n",
       "      <td>0.107839</td>\n",
       "      <td>0.063127</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM813290</th>\n",
       "      <td>0.773797</td>\n",
       "      <td>0.526281</td>\n",
       "      <td>0.07592</td>\n",
       "      <td>0.233635</td>\n",
       "      <td>0.046912</td>\n",
       "      <td>0.04386</td>\n",
       "      <td>0.982608</td>\n",
       "      <td>0.022921</td>\n",
       "      <td>0.021091</td>\n",
       "      <td>0.709384</td>\n",
       "      <td>...</td>\n",
       "      <td>0.695083</td>\n",
       "      <td>0.154565</td>\n",
       "      <td>0.852972</td>\n",
       "      <td>0.077501</td>\n",
       "      <td>0.156859</td>\n",
       "      <td>0.03269</td>\n",
       "      <td>0.387466</td>\n",
       "      <td>0.019694</td>\n",
       "      <td>0.047126</td>\n",
       "      <td>0.046525</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM813291</th>\n",
       "      <td>0.4425</td>\n",
       "      <td>0.367014</td>\n",
       "      <td>0.109204</td>\n",
       "      <td>0.503786</td>\n",
       "      <td>0.028654</td>\n",
       "      <td>0.034655</td>\n",
       "      <td>0.807651</td>\n",
       "      <td>0.023715</td>\n",
       "      <td>0.018089</td>\n",
       "      <td>0.653145</td>\n",
       "      <td>...</td>\n",
       "      <td>0.713714</td>\n",
       "      <td>0.10791</td>\n",
       "      <td>0.450145</td>\n",
       "      <td>0.056143</td>\n",
       "      <td>0.142126</td>\n",
       "      <td>0.055615</td>\n",
       "      <td>0.493</td>\n",
       "      <td>0.009382</td>\n",
       "      <td>0.039019</td>\n",
       "      <td>0.033644</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>118 rows × 26300 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "ID_REF    cg00000292 cg00002426 cg00003994 cg00005847 cg00006414 cg00007981  \\\n",
       "GSM813174   0.619946   0.460933    0.05196   0.240433   0.013831   0.023905   \n",
       "GSM813175    0.42915   0.361039   0.066462   0.712471   0.105384   0.146959   \n",
       "GSM813176   0.598473   0.399592   0.081039    0.19372   0.039534   0.024417   \n",
       "GSM813177   0.605416   0.596668   0.137904   0.241135   0.035636   0.015306   \n",
       "GSM813178   0.587209   0.436555   0.049907   0.179162   0.017098    0.02515   \n",
       "...              ...        ...        ...        ...        ...        ...   \n",
       "GSM813287    0.58477   0.608247    0.17715   0.386829   0.024344    0.03662   \n",
       "GSM813288   0.637604   0.413775    0.06929   0.211992    0.01932   0.026035   \n",
       "GSM813289   0.624775   0.689305   0.305079   0.516924   0.150818     0.1096   \n",
       "GSM813290   0.773797   0.526281    0.07592   0.233635   0.046912    0.04386   \n",
       "GSM813291     0.4425   0.367014   0.109204   0.503786   0.028654   0.034655   \n",
       "\n",
       "ID_REF    cg00008493 cg00008713 cg00009407 cg00010193  ... cg27653134  \\\n",
       "GSM813174   0.990165   0.014497   0.006952   0.662304  ...   0.695425   \n",
       "GSM813175   0.988713   0.026515   0.031751   0.588816  ...   0.619898   \n",
       "GSM813176   0.984537   0.020919   0.013639   0.663489  ...   0.826355   \n",
       "GSM813177   0.988111   0.027033   0.008842   0.675272  ...   0.796165   \n",
       "GSM813178   0.990127   0.007726   0.008732   0.603307  ...   0.704033   \n",
       "...              ...        ...        ...        ...  ...        ...   \n",
       "GSM813287   0.976659   0.010312   0.007879    0.64371  ...   0.458203   \n",
       "GSM813288   0.986941   0.012445   0.011437   0.680988  ...   0.693108   \n",
       "GSM813289   0.952523   0.037228   0.043694   0.660779  ...   0.582403   \n",
       "GSM813290   0.982608   0.022921   0.021091   0.709384  ...   0.695083   \n",
       "GSM813291   0.807651   0.023715   0.018089   0.653145  ...   0.713714   \n",
       "\n",
       "ID_REF    cg27654142 cg27655855 cg27655905 cg27657249 cg27657283 cg27661264  \\\n",
       "GSM813174   0.093868   0.816186   0.033004   0.170945   0.034743   0.370746   \n",
       "GSM813175   0.054866   0.148381   0.083458   0.102413   0.137931   0.339054   \n",
       "GSM813176   0.053182   0.800741   0.045924   0.137161   0.052218   0.396524   \n",
       "GSM813177   0.097044   0.812453   0.050538   0.174674   0.031456   0.292604   \n",
       "GSM813178   0.076322   0.828708   0.031439   0.202882   0.035126   0.261081   \n",
       "...              ...        ...        ...        ...        ...        ...   \n",
       "GSM813287   0.032158   0.677973   0.031384   0.156193   0.054776   0.395179   \n",
       "GSM813288    0.03265   0.772188   0.034177    0.18395   0.048791   0.301718   \n",
       "GSM813289   0.165378   0.763463   0.124898   0.179512    0.15737   0.444144   \n",
       "GSM813290   0.154565   0.852972   0.077501   0.156859    0.03269   0.387466   \n",
       "GSM813291    0.10791   0.450145   0.056143   0.142126   0.055615      0.493   \n",
       "\n",
       "ID_REF    cg27662379 cg27662877 cg27665659  \n",
       "GSM813174   0.006512     0.0124   0.066593  \n",
       "GSM813175   0.041294   0.199616    0.03218  \n",
       "GSM813176   0.012065   0.023331   0.059368  \n",
       "GSM813177   0.009658   0.026601   0.037425  \n",
       "GSM813178   0.009629   0.013818   0.060106  \n",
       "...              ...        ...        ...  \n",
       "GSM813287   0.013456   0.035263   0.054444  \n",
       "GSM813288    0.00785   0.017347   0.068875  \n",
       "GSM813289    0.06769   0.107839   0.063127  \n",
       "GSM813290   0.019694   0.047126   0.046525  \n",
       "GSM813291   0.009382   0.039019   0.033644  \n",
       "\n",
       "[118 rows x 26300 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from sklearn.preprocessing import LabelEncoder\n",
    "\n",
    "encoder = LabelEncoder()\n",
    "\n",
    "data_matrix_nf[\"tissue_char\"] = encoder.fit_transform(data_matrix_nf[\"tissue_char\"])\n",
    "\n",
    "data_matrix_nf.iloc[:,:-1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "1ba01701",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "01-May-2023 16:02:05 INFO GEOparse - Downloading ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE63nnn/GSE63384/soft/GSE63384_family.soft.gz to /bin/LungCancerTest\\GSE63384_family.soft.gz\n",
      "100%|██████████| 20.3M/20.3M [00:07<00:00, 2.90MB/s]\n",
      "01-May-2023 16:02:16 DEBUG downloader - Size validation passed\n",
      "01-May-2023 16:02:16 DEBUG downloader - Moving C:\\Users\\FRIEND~1.DES\\AppData\\Local\\Temp\\tmpore_ktzx to C:\\bin\\LungCancerTest\\GSE63384_family.soft.gz\n",
      "01-May-2023 16:02:16 DEBUG downloader - Successfully downloaded ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE63nnn/GSE63384/soft/GSE63384_family.soft.gz\n",
      "01-May-2023 16:02:16 INFO GEOparse - Parsing /bin/LungCancerTest\\GSE63384_family.soft.gz: \n",
      "01-May-2023 16:02:16 DEBUG GEOparse - DATABASE: GeoMiame\n",
      "01-May-2023 16:02:16 DEBUG GEOparse - SERIES: GSE63384\n",
      "01-May-2023 16:02:16 DEBUG GEOparse - PLATFORM: GPL8490\n",
      "01-May-2023 16:02:17 DEBUG GEOparse - SAMPLE: GSM1547818\n",
      "01-May-2023 16:02:17 DEBUG GEOparse - SAMPLE: GSM1547819\n",
      "01-May-2023 16:02:17 DEBUG GEOparse - SAMPLE: GSM1547820\n",
      "01-May-2023 16:02:17 DEBUG GEOparse - SAMPLE: GSM1547821\n",
      "01-May-2023 16:02:17 DEBUG GEOparse - SAMPLE: GSM1547822\n",
      "01-May-2023 16:02:17 DEBUG GEOparse - SAMPLE: GSM1547823\n",
      "01-May-2023 16:02:17 DEBUG GEOparse - SAMPLE: GSM1547824\n",
      "01-May-2023 16:02:18 DEBUG GEOparse - SAMPLE: GSM1547825\n",
      "01-May-2023 16:02:18 DEBUG GEOparse - SAMPLE: GSM1547826\n",
      "01-May-2023 16:02:18 DEBUG GEOparse - SAMPLE: GSM1547827\n",
      "01-May-2023 16:02:18 DEBUG GEOparse - SAMPLE: GSM1547828\n",
      "01-May-2023 16:02:18 DEBUG GEOparse - SAMPLE: GSM1547829\n",
      "01-May-2023 16:02:18 DEBUG GEOparse - SAMPLE: GSM1547830\n",
      "01-May-2023 16:02:18 DEBUG GEOparse - SAMPLE: GSM1547831\n",
      "01-May-2023 16:02:19 DEBUG GEOparse - SAMPLE: GSM1547832\n",
      "01-May-2023 16:02:19 DEBUG GEOparse - SAMPLE: GSM1547833\n",
      "01-May-2023 16:02:19 DEBUG GEOparse - SAMPLE: GSM1547834\n",
      "01-May-2023 16:02:19 DEBUG GEOparse - SAMPLE: GSM1547835\n",
      "01-May-2023 16:02:19 DEBUG GEOparse - SAMPLE: GSM1547836\n",
      "01-May-2023 16:02:19 DEBUG GEOparse - SAMPLE: GSM1547837\n",
      "01-May-2023 16:02:20 DEBUG GEOparse - SAMPLE: GSM1547838\n",
      "01-May-2023 16:02:20 DEBUG GEOparse - SAMPLE: GSM1547839\n",
      "01-May-2023 16:02:20 DEBUG GEOparse - SAMPLE: GSM1547840\n",
      "01-May-2023 16:02:20 DEBUG GEOparse - SAMPLE: GSM1547841\n",
      "01-May-2023 16:02:20 DEBUG GEOparse - SAMPLE: GSM1547842\n",
      "01-May-2023 16:02:20 DEBUG GEOparse - SAMPLE: GSM1547843\n",
      "01-May-2023 16:02:20 DEBUG GEOparse - SAMPLE: GSM1547844\n",
      "01-May-2023 16:02:20 DEBUG GEOparse - SAMPLE: GSM1547845\n",
      "01-May-2023 16:02:21 DEBUG GEOparse - SAMPLE: GSM1547846\n",
      "01-May-2023 16:02:21 DEBUG GEOparse - SAMPLE: GSM1547847\n",
      "01-May-2023 16:02:21 DEBUG GEOparse - SAMPLE: GSM1547848\n",
      "01-May-2023 16:02:21 DEBUG GEOparse - SAMPLE: GSM1547849\n",
      "01-May-2023 16:02:21 DEBUG GEOparse - SAMPLE: GSM1547850\n",
      "01-May-2023 16:02:21 DEBUG GEOparse - SAMPLE: GSM1547851\n",
      "01-May-2023 16:02:21 DEBUG GEOparse - SAMPLE: GSM1547852\n",
      "01-May-2023 16:02:22 DEBUG GEOparse - SAMPLE: GSM1547853\n",
      "01-May-2023 16:02:22 DEBUG GEOparse - SAMPLE: GSM1547854\n",
      "01-May-2023 16:02:22 DEBUG GEOparse - SAMPLE: GSM1547855\n",
      "01-May-2023 16:02:22 DEBUG GEOparse - SAMPLE: GSM1547856\n",
      "01-May-2023 16:02:22 DEBUG GEOparse - SAMPLE: GSM1547857\n",
      "01-May-2023 16:02:23 DEBUG GEOparse - SAMPLE: GSM1547858\n",
      "01-May-2023 16:02:23 DEBUG GEOparse - SAMPLE: GSM1547859\n",
      "01-May-2023 16:02:23 DEBUG GEOparse - SAMPLE: GSM1547860\n",
      "01-May-2023 16:02:23 DEBUG GEOparse - SAMPLE: GSM1547861\n",
      "01-May-2023 16:02:23 DEBUG GEOparse - SAMPLE: GSM1547862\n",
      "01-May-2023 16:02:23 DEBUG GEOparse - SAMPLE: GSM1547863\n",
      "01-May-2023 16:02:23 DEBUG GEOparse - SAMPLE: GSM1547864\n",
      "01-May-2023 16:02:23 DEBUG GEOparse - SAMPLE: GSM1547865\n",
      "01-May-2023 16:02:23 DEBUG GEOparse - SAMPLE: GSM1547866\n",
      "01-May-2023 16:02:24 DEBUG GEOparse - SAMPLE: GSM1547867\n",
      "01-May-2023 16:02:24 DEBUG GEOparse - SAMPLE: GSM1547868\n",
      "01-May-2023 16:02:24 DEBUG GEOparse - SAMPLE: GSM1547869\n",
      "01-May-2023 16:02:24 DEBUG GEOparse - SAMPLE: GSM1547870\n",
      "01-May-2023 16:02:24 DEBUG GEOparse - SAMPLE: GSM1547871\n",
      "01-May-2023 16:02:24 DEBUG GEOparse - SAMPLE: GSM1547872\n",
      "01-May-2023 16:02:24 DEBUG GEOparse - SAMPLE: GSM1547873\n",
      "01-May-2023 16:02:24 DEBUG GEOparse - SAMPLE: GSM1547874\n",
      "01-May-2023 16:02:24 DEBUG GEOparse - SAMPLE: GSM1547875\n",
      "01-May-2023 16:02:25 DEBUG GEOparse - SAMPLE: GSM1547876\n",
      "01-May-2023 16:02:25 DEBUG GEOparse - SAMPLE: GSM1547877\n",
      "01-May-2023 16:02:25 DEBUG GEOparse - SAMPLE: GSM1547878\n",
      "01-May-2023 16:02:25 DEBUG GEOparse - SAMPLE: GSM1547879\n",
      "01-May-2023 16:02:25 DEBUG GEOparse - SAMPLE: GSM1547880\n",
      "01-May-2023 16:02:25 DEBUG GEOparse - SAMPLE: GSM1547881\n",
      "01-May-2023 16:02:25 DEBUG GEOparse - SAMPLE: GSM1547882\n",
      "01-May-2023 16:02:25 DEBUG GEOparse - SAMPLE: GSM1547883\n",
      "01-May-2023 16:02:26 DEBUG GEOparse - SAMPLE: GSM1547884\n",
      "01-May-2023 16:02:26 DEBUG GEOparse - SAMPLE: GSM1547885\n",
      "01-May-2023 16:02:26 DEBUG GEOparse - SAMPLE: GSM1547886\n",
      "01-May-2023 16:02:26 DEBUG GEOparse - SAMPLE: GSM1547887\n"
     ]
    }
   ],
   "source": [
    "gse_test = GEOparse.get_GEO(geo=\"GSE63384\", destdir=\"/bin/LungCancerTest\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "1eda5882",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "GSM example:\n",
      "Name:  GSM1547818\n",
      "Metadata:\n",
      " - title : lung, nontumor adjacent_12839\n",
      " - geo_accession : GSM1547818\n",
      " - status : Public on Apr 30 2015\n",
      " - submission_date : Nov 17 2014\n",
      " - last_update_date : Apr 30 2015\n",
      " - type : genomic\n",
      " - channel_count : 1\n",
      " - source_name_ch1 : lung, nontumor adjacent\n",
      " - organism_ch1 : Homo sapiens\n",
      " - taxid_ch1 : 9606\n",
      " - characteristics_ch1 : tissue: lung, patient_id1: 12839N, patient_id: 12839, analysis batch: 2, Sex: Female, race: European American, age: 62, disease state: non-small cell lung cancer (NSCLC), histology: normal, smoking status: ever, smoking packyears: 65, death due to cancer: no, survival after surgery (days): 3876\n",
      " - molecule_ch1 : genomic DNA\n",
      " - extract_protocol_ch1 : genomic DNA was extracted and purified from tissue samples using Qiagen DNeasy Kit according to standard instructions\n",
      " - label_ch1 : Cy5 and Cy3\n",
      " - label_protocol_ch1 : Standard Illumina Protocol\n",
      " - hyb_protocol : bisulphite converted DNA was amplified, fragmented and hybridised to Illumina Infinium Human Methylation27 Beadchip using standard Illumina protocol\n",
      " - scan_protocol : Arrays were imaged using BeadArray Reader using standard recommended Illumina scanner setting\n",
      " - description : nontumor adjacent\n",
      " - data_processing : BeadStudio software v3.2; Fluorescent signals were collated via Bead Studio software and converted into -values. Illumina Bead Studio software output data were normalized using SSN (simple scaling normalization), to correct for dye bias and transformed into M-values (M = log2 intensity of methylated probe/ intensity of unmethylated probe), using the R package lumi. Raw signals can be found on the series level in the file Unmethylated_and_methylated_signal_intensities.txt\n",
      " - platform_id : GPL8490\n",
      " - contact_name : Ana,I,Robles\n",
      " - contact_email : Ana_Robles@nih.gov\n",
      " - contact_phone : 301-496-1729\n",
      " - contact_laboratory : Laboratory of Human Carcinogenesis\n",
      " - contact_institute : National Cancer Institute/NIH\n",
      " - contact_address : 37 Convent Dr 37/3060D\n",
      " - contact_city : Bethesda\n",
      " - contact_state : MD\n",
      " - contact_zip/postal_code : 20892\n",
      " - contact_country : USA\n",
      " - supplementary_file : NONE\n",
      " - series_id : GSE63384\n",
      " - data_row_count : 27578\n",
      "Table data:\n",
      "       ID_REF     VALUE\n",
      "0  cg00000292  0.677262\n",
      "1  cg00002426  0.418543\n",
      "2  cg00003994  0.035070\n",
      "3  cg00005847  0.214901\n",
      "4  cg00006414  0.038121\n",
      "\n",
      "GPL example:\n",
      "Name:  GPL8490\n",
      "Metadata:\n",
      " - title : Illumina HumanMethylation27 BeadChip (HumanMethylation27_270596_v.1.2)\n",
      " - geo_accession : GPL8490\n",
      " - status : Public on Apr 27 2009\n",
      " - submission_date : Apr 27 2009\n",
      " - last_update_date : Jan 02 2015\n",
      " - technology : oligonucleotide beads\n",
      " - distribution : commercial\n",
      " - organism : Homo sapiens\n",
      " - taxid : 9606\n",
      " - manufacturer : Illumina, Inc.\n",
      " - manufacture_protocol : See manufacturer's website\n",
      " - description : HumanMethylation27 DNA Analysis BeadChip allows researchers to interrogate 27,578 highly informative CpG sites per sample at single-nucleotide resolution. This 12-sample BeadChip features content derived from the well-annotated NCBI CCDS database (Genome Build 36) and is supplemented with more than 1,000 cancer-related genes described in published literature. Probe content has been enriched to deeply cover more than 150 well-established cancer genes known to show differential methylation patterns. HumanMethylation27 BeadChip content also targets the promoter regions of 110 miRNA genes.\n",
      " - web_link : http://www.illumina.com/pages.ilmn?ID=243\n",
      " - contact_name : ,,Illumina Inc.\n",
      " - contact_email : expression@illumina.com, techsupport@illumina.com\n",
      " - contact_phone : 1 800 809 4566\n",
      " - contact_institute : Illumina Inc.\n",
      " - contact_address : 9885 Towne Centre Drive\n",
      " - contact_city : San Diego\n",
      " - contact_state : CA\n",
      " - contact_zip/postal_code : 92121\n",
      " - contact_country : USA\n",
      " - contact_web_link : www.illumina.com\n",
      " - supplementary_file : ftp://ftp.ncbi.nlm.nih.gov/geo/platforms/GPL8nnn/GPL8490/suppl/GPL8490_HumanMethylation27_270596_v.1.2.csv.gz\n",
      " - data_row_count : 27578\n",
      "Table data:\n",
      "           ID        Name IlmnStrand  AddressA_ID  \\\n",
      "0  cg00000292  cg00000292        TOP       990370   \n",
      "1  cg00002426  cg00002426        TOP      6580397   \n",
      "2  cg00003994  cg00003994        TOP      7150184   \n",
      "3  cg00005847  cg00005847        BOT      4850717   \n",
      "4  cg00006414  cg00006414        BOT      6980731   \n",
      "\n",
      "                                    AlleleA_ProbeSeq  AddressB_ID  \\\n",
      "0  AAACATTAATTACCAACCACTCTTCCAAAAAACACTTACCATTAAA...      6660678   \n",
      "1  AATATAATAACATTACCTTACCCATCTTATAATCAAACCAAACAAA...      6100343   \n",
      "2  AATAATAATAATACCCCCTATAATACTAACTAACAAACATACCCTC...      7150392   \n",
      "3  TACTATAATACACCCTATATTTAAAACACTAAACTTACCCCATTAA...      1260113   \n",
      "4  CTCAAAAACCAAACAAAACAAAACCCCAATACTAATCATTAATAAA...      4280093   \n",
      "\n",
      "                                    AlleleB_ProbeSeq  GenomeBuild Chr  \\\n",
      "0  AAACATTAATTACCAACCGCTCTTCCAAAAAACACTTACCATTAAA...           36  16   \n",
      "1  AATATAATAACATTACCTTACCCGTCTTATAATCAAACCAAACGAA...           36   3   \n",
      "2  AATAATAATAATACCCCCTATAATACTAACTAACAAACATACCCTC...           36   7   \n",
      "3  TACTATAATACACCCTATATTTAAAACACTAAACTTACCCCATTAA...           36   2   \n",
      "4  CTCGAAAACCGAACAAAACAAAACCCCAATACTAATCGTTAATAAA...           36   7   \n",
      "\n",
      "     MapInfo  ... Distance_to_TSS CPG_ISLAND   CPG_ISLAND_LOCATIONS  \\\n",
      "0   28797601  ...           291.0       True   16:28797486-28797825   \n",
      "1   57718583  ...           369.0       True    3:57716811-57718675   \n",
      "2   15692387  ...           432.0       True    7:15691512-15693551   \n",
      "3  176737319  ...           268.0      False                    NaN   \n",
      "4  148453770  ...           671.0       True  7:148453584-148455804   \n",
      "\n",
      "   MIR_CPG_ISLAND      RANGE_GB  RANGE_START    RANGE_END RANGE_STRAND  \\\n",
      "0             NaN   NC_000016.8   28797486.0   28797825.0            +   \n",
      "1             NaN  NC_000003.10   57716811.0   57718675.0            +   \n",
      "2             NaN  NC_000007.12   15691512.0   15693551.0            -   \n",
      "3             NaN           NaN          NaN          NaN          NaN   \n",
      "4             NaN  NC_000007.12  148453584.0  148455804.0            +   \n",
      "\n",
      "        GB_ACC    ORF  \n",
      "0  NM_173201.2    487  \n",
      "1  NM_007159.2   7871  \n",
      "2  NM_005924.3   4223  \n",
      "3  NM_006898.4   3232  \n",
      "4  NM_020781.2  57541  \n",
      "\n",
      "[5 rows x 38 columns]\n"
     ]
    }
   ],
   "source": [
    "print()\n",
    "print(\"GSM example:\")\n",
    "for gsm_name, gsm in gse_test.gsms.items():\n",
    "    print(\"Name: \", gsm_name)\n",
    "    print(\"Metadata:\",)\n",
    "    for key, value in gsm.metadata.items():\n",
    "        print(\" - %s : %s\" % (key, \", \".join(value)))\n",
    "    print (\"Table data:\",)\n",
    "    print (gsm.table.head())\n",
    "    break\n",
    "\n",
    "print()\n",
    "print(\"GPL example:\")\n",
    "for gpl_name, gpl in gse_test.gpls.items():\n",
    "    print(\"Name: \", gpl_name)\n",
    "    print(\"Metadata:\",)\n",
    "    for key, value in gpl.metadata.items():\n",
    "        print(\" - %s : %s\" % (key, \", \".join(value)))\n",
    "    print(\"Table data:\",)\n",
    "    print(gpl.table.head())\n",
    "    break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "6473988e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "tissue_test = gse_test.phenotype_data['source_name_ch1']\n",
    "tissue_test = tissue_test.rename(\"tissue_char\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "e79bcd3c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\FRIENDS.DESKTOP-1RIRFCP\\AppData\\Local\\Temp\\ipykernel_7160\\4075013598.py:2: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  test_data_matrix = test_data_matrix.append(tissue_test)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>ID_REF</th>\n",
       "      <th>cg00000292</th>\n",
       "      <th>cg00002426</th>\n",
       "      <th>cg00003994</th>\n",
       "      <th>cg00005847</th>\n",
       "      <th>cg00006414</th>\n",
       "      <th>cg00007981</th>\n",
       "      <th>cg00008493</th>\n",
       "      <th>cg00008713</th>\n",
       "      <th>cg00009407</th>\n",
       "      <th>cg00010193</th>\n",
       "      <th>...</th>\n",
       "      <th>cg27654142</th>\n",
       "      <th>cg27655855</th>\n",
       "      <th>cg27655905</th>\n",
       "      <th>cg27657249</th>\n",
       "      <th>cg27657283</th>\n",
       "      <th>cg27661264</th>\n",
       "      <th>cg27662379</th>\n",
       "      <th>cg27662877</th>\n",
       "      <th>cg27665659</th>\n",
       "      <th>tissue_char</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GSM1547818</th>\n",
       "      <td>0.677262</td>\n",
       "      <td>0.418543</td>\n",
       "      <td>0.03507</td>\n",
       "      <td>0.214901</td>\n",
       "      <td>0.038121</td>\n",
       "      <td>0.048343</td>\n",
       "      <td>0.984074</td>\n",
       "      <td>0.017477</td>\n",
       "      <td>0.010621</td>\n",
       "      <td>0.687097</td>\n",
       "      <td>...</td>\n",
       "      <td>0.039206</td>\n",
       "      <td>0.818921</td>\n",
       "      <td>0.058349</td>\n",
       "      <td>0.287538</td>\n",
       "      <td>0.028253</td>\n",
       "      <td>0.245344</td>\n",
       "      <td>0.018762</td>\n",
       "      <td>0.025226</td>\n",
       "      <td>0.051665</td>\n",
       "      <td>lung, nontumor adjacent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM1547819</th>\n",
       "      <td>0.386802</td>\n",
       "      <td>0.292361</td>\n",
       "      <td>0.028753</td>\n",
       "      <td>0.316076</td>\n",
       "      <td>0.033974</td>\n",
       "      <td>0.07034</td>\n",
       "      <td>0.98456</td>\n",
       "      <td>0.016491</td>\n",
       "      <td>0.010838</td>\n",
       "      <td>0.643857</td>\n",
       "      <td>...</td>\n",
       "      <td>0.048519</td>\n",
       "      <td>0.50977</td>\n",
       "      <td>0.059878</td>\n",
       "      <td>0.258576</td>\n",
       "      <td>0.022874</td>\n",
       "      <td>0.196758</td>\n",
       "      <td>0.015803</td>\n",
       "      <td>0.014159</td>\n",
       "      <td>0.047561</td>\n",
       "      <td>lung, adenocarcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM1547820</th>\n",
       "      <td>0.523519</td>\n",
       "      <td>0.2532</td>\n",
       "      <td>0.04043</td>\n",
       "      <td>0.214164</td>\n",
       "      <td>0.043849</td>\n",
       "      <td>0.047455</td>\n",
       "      <td>0.976319</td>\n",
       "      <td>0.017809</td>\n",
       "      <td>0.010469</td>\n",
       "      <td>0.672972</td>\n",
       "      <td>...</td>\n",
       "      <td>0.034228</td>\n",
       "      <td>0.764041</td>\n",
       "      <td>0.053957</td>\n",
       "      <td>0.263101</td>\n",
       "      <td>0.025725</td>\n",
       "      <td>0.175875</td>\n",
       "      <td>0.017959</td>\n",
       "      <td>0.027677</td>\n",
       "      <td>0.074324</td>\n",
       "      <td>lung, nontumor adjacent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM1547821</th>\n",
       "      <td>0.277912</td>\n",
       "      <td>0.836009</td>\n",
       "      <td>0.036721</td>\n",
       "      <td>0.308525</td>\n",
       "      <td>0.025322</td>\n",
       "      <td>0.090031</td>\n",
       "      <td>0.978845</td>\n",
       "      <td>0.015602</td>\n",
       "      <td>0.005646</td>\n",
       "      <td>0.671881</td>\n",
       "      <td>...</td>\n",
       "      <td>0.043453</td>\n",
       "      <td>0.845068</td>\n",
       "      <td>0.044641</td>\n",
       "      <td>0.308361</td>\n",
       "      <td>0.021629</td>\n",
       "      <td>0.154793</td>\n",
       "      <td>0.019908</td>\n",
       "      <td>0.025395</td>\n",
       "      <td>0.041205</td>\n",
       "      <td>lung, adenocarcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM1547822</th>\n",
       "      <td>0.633324</td>\n",
       "      <td>0.435435</td>\n",
       "      <td>0.033689</td>\n",
       "      <td>0.264442</td>\n",
       "      <td>0.055146</td>\n",
       "      <td>0.037193</td>\n",
       "      <td>0.97957</td>\n",
       "      <td>0.017041</td>\n",
       "      <td>0.011099</td>\n",
       "      <td>0.640128</td>\n",
       "      <td>...</td>\n",
       "      <td>0.045588</td>\n",
       "      <td>0.770388</td>\n",
       "      <td>0.057259</td>\n",
       "      <td>0.258144</td>\n",
       "      <td>0.021643</td>\n",
       "      <td>0.218276</td>\n",
       "      <td>0.017326</td>\n",
       "      <td>0.024316</td>\n",
       "      <td>0.077373</td>\n",
       "      <td>lung, nontumor adjacent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM1547883</th>\n",
       "      <td>0.699218</td>\n",
       "      <td>0.466178</td>\n",
       "      <td>0.023914</td>\n",
       "      <td>0.293032</td>\n",
       "      <td>0.047161</td>\n",
       "      <td>0.14833</td>\n",
       "      <td>0.978287</td>\n",
       "      <td>0.01298</td>\n",
       "      <td>0.00742</td>\n",
       "      <td>0.660226</td>\n",
       "      <td>...</td>\n",
       "      <td>0.058156</td>\n",
       "      <td>0.783083</td>\n",
       "      <td>0.060586</td>\n",
       "      <td>0.248578</td>\n",
       "      <td>0.016919</td>\n",
       "      <td>0.202043</td>\n",
       "      <td>0.015655</td>\n",
       "      <td>0.027396</td>\n",
       "      <td>0.041758</td>\n",
       "      <td>lung, adenocarcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM1547884</th>\n",
       "      <td>0.68797</td>\n",
       "      <td>0.491488</td>\n",
       "      <td>0.039781</td>\n",
       "      <td>0.203612</td>\n",
       "      <td>0.046104</td>\n",
       "      <td>0.085663</td>\n",
       "      <td>0.979978</td>\n",
       "      <td>0.039501</td>\n",
       "      <td>0.009376</td>\n",
       "      <td>0.638249</td>\n",
       "      <td>...</td>\n",
       "      <td>0.057821</td>\n",
       "      <td>0.848221</td>\n",
       "      <td>0.144379</td>\n",
       "      <td>0.421445</td>\n",
       "      <td>0.04819</td>\n",
       "      <td>0.322788</td>\n",
       "      <td>0.015859</td>\n",
       "      <td>0.022922</td>\n",
       "      <td>0.145287</td>\n",
       "      <td>lung, nontumor adjacent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM1547885</th>\n",
       "      <td>0.795324</td>\n",
       "      <td>0.422477</td>\n",
       "      <td>0.03757</td>\n",
       "      <td>0.354545</td>\n",
       "      <td>0.035515</td>\n",
       "      <td>0.128453</td>\n",
       "      <td>0.986352</td>\n",
       "      <td>0.028905</td>\n",
       "      <td>0.008717</td>\n",
       "      <td>0.629348</td>\n",
       "      <td>...</td>\n",
       "      <td>0.095023</td>\n",
       "      <td>0.762515</td>\n",
       "      <td>0.144673</td>\n",
       "      <td>0.382796</td>\n",
       "      <td>0.033938</td>\n",
       "      <td>0.281695</td>\n",
       "      <td>0.018043</td>\n",
       "      <td>0.024738</td>\n",
       "      <td>0.076221</td>\n",
       "      <td>lung, adenocarcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM1547886</th>\n",
       "      <td>0.728728</td>\n",
       "      <td>0.552402</td>\n",
       "      <td>0.014772</td>\n",
       "      <td>0.197786</td>\n",
       "      <td>0.063125</td>\n",
       "      <td>0.126063</td>\n",
       "      <td>0.983295</td>\n",
       "      <td>0.018053</td>\n",
       "      <td>0.0066</td>\n",
       "      <td>0.70563</td>\n",
       "      <td>...</td>\n",
       "      <td>0.046783</td>\n",
       "      <td>0.785088</td>\n",
       "      <td>0.05279</td>\n",
       "      <td>0.254532</td>\n",
       "      <td>0.033974</td>\n",
       "      <td>0.282958</td>\n",
       "      <td>0.01129</td>\n",
       "      <td>0.031366</td>\n",
       "      <td>0.081158</td>\n",
       "      <td>lung, nontumor adjacent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM1547887</th>\n",
       "      <td>0.729681</td>\n",
       "      <td>0.375562</td>\n",
       "      <td>0.029851</td>\n",
       "      <td>0.20372</td>\n",
       "      <td>0.062058</td>\n",
       "      <td>0.06201</td>\n",
       "      <td>0.975873</td>\n",
       "      <td>0.015436</td>\n",
       "      <td>0.010966</td>\n",
       "      <td>0.696899</td>\n",
       "      <td>...</td>\n",
       "      <td>0.145468</td>\n",
       "      <td>0.712126</td>\n",
       "      <td>0.076845</td>\n",
       "      <td>0.236373</td>\n",
       "      <td>0.024206</td>\n",
       "      <td>0.351849</td>\n",
       "      <td>0.020035</td>\n",
       "      <td>0.03603</td>\n",
       "      <td>0.046374</td>\n",
       "      <td>lung, adenocarcinoma</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>70 rows × 27579 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "ID_REF     cg00000292 cg00002426 cg00003994 cg00005847 cg00006414 cg00007981  \\\n",
       "GSM1547818   0.677262   0.418543    0.03507   0.214901   0.038121   0.048343   \n",
       "GSM1547819   0.386802   0.292361   0.028753   0.316076   0.033974    0.07034   \n",
       "GSM1547820   0.523519     0.2532    0.04043   0.214164   0.043849   0.047455   \n",
       "GSM1547821   0.277912   0.836009   0.036721   0.308525   0.025322   0.090031   \n",
       "GSM1547822   0.633324   0.435435   0.033689   0.264442   0.055146   0.037193   \n",
       "...               ...        ...        ...        ...        ...        ...   \n",
       "GSM1547883   0.699218   0.466178   0.023914   0.293032   0.047161    0.14833   \n",
       "GSM1547884    0.68797   0.491488   0.039781   0.203612   0.046104   0.085663   \n",
       "GSM1547885   0.795324   0.422477    0.03757   0.354545   0.035515   0.128453   \n",
       "GSM1547886   0.728728   0.552402   0.014772   0.197786   0.063125   0.126063   \n",
       "GSM1547887   0.729681   0.375562   0.029851    0.20372   0.062058    0.06201   \n",
       "\n",
       "ID_REF     cg00008493 cg00008713 cg00009407 cg00010193  ... cg27654142  \\\n",
       "GSM1547818   0.984074   0.017477   0.010621   0.687097  ...   0.039206   \n",
       "GSM1547819    0.98456   0.016491   0.010838   0.643857  ...   0.048519   \n",
       "GSM1547820   0.976319   0.017809   0.010469   0.672972  ...   0.034228   \n",
       "GSM1547821   0.978845   0.015602   0.005646   0.671881  ...   0.043453   \n",
       "GSM1547822    0.97957   0.017041   0.011099   0.640128  ...   0.045588   \n",
       "...               ...        ...        ...        ...  ...        ...   \n",
       "GSM1547883   0.978287    0.01298    0.00742   0.660226  ...   0.058156   \n",
       "GSM1547884   0.979978   0.039501   0.009376   0.638249  ...   0.057821   \n",
       "GSM1547885   0.986352   0.028905   0.008717   0.629348  ...   0.095023   \n",
       "GSM1547886   0.983295   0.018053     0.0066    0.70563  ...   0.046783   \n",
       "GSM1547887   0.975873   0.015436   0.010966   0.696899  ...   0.145468   \n",
       "\n",
       "ID_REF     cg27655855 cg27655905 cg27657249 cg27657283 cg27661264 cg27662379  \\\n",
       "GSM1547818   0.818921   0.058349   0.287538   0.028253   0.245344   0.018762   \n",
       "GSM1547819    0.50977   0.059878   0.258576   0.022874   0.196758   0.015803   \n",
       "GSM1547820   0.764041   0.053957   0.263101   0.025725   0.175875   0.017959   \n",
       "GSM1547821   0.845068   0.044641   0.308361   0.021629   0.154793   0.019908   \n",
       "GSM1547822   0.770388   0.057259   0.258144   0.021643   0.218276   0.017326   \n",
       "...               ...        ...        ...        ...        ...        ...   \n",
       "GSM1547883   0.783083   0.060586   0.248578   0.016919   0.202043   0.015655   \n",
       "GSM1547884   0.848221   0.144379   0.421445    0.04819   0.322788   0.015859   \n",
       "GSM1547885   0.762515   0.144673   0.382796   0.033938   0.281695   0.018043   \n",
       "GSM1547886   0.785088    0.05279   0.254532   0.033974   0.282958    0.01129   \n",
       "GSM1547887   0.712126   0.076845   0.236373   0.024206   0.351849   0.020035   \n",
       "\n",
       "ID_REF     cg27662877 cg27665659              tissue_char  \n",
       "GSM1547818   0.025226   0.051665  lung, nontumor adjacent  \n",
       "GSM1547819   0.014159   0.047561     lung, adenocarcinoma  \n",
       "GSM1547820   0.027677   0.074324  lung, nontumor adjacent  \n",
       "GSM1547821   0.025395   0.041205     lung, adenocarcinoma  \n",
       "GSM1547822   0.024316   0.077373  lung, nontumor adjacent  \n",
       "...               ...        ...                      ...  \n",
       "GSM1547883   0.027396   0.041758     lung, adenocarcinoma  \n",
       "GSM1547884   0.022922   0.145287  lung, nontumor adjacent  \n",
       "GSM1547885   0.024738   0.076221     lung, adenocarcinoma  \n",
       "GSM1547886   0.031366   0.081158  lung, nontumor adjacent  \n",
       "GSM1547887    0.03603   0.046374     lung, adenocarcinoma  \n",
       "\n",
       "[70 rows x 27579 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_data_matrix = gse_test.pivot_samples('VALUE')\n",
    "test_data_matrix = test_data_matrix.append(tissue_test)\n",
    "test_data_matrix = test_data_matrix.transpose()\n",
    "test_data_matrix = test_data_matrix.dropna()\n",
    "test_data_matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "2604f930",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>ID_REF</th>\n",
       "      <th>cg00000292</th>\n",
       "      <th>cg00002426</th>\n",
       "      <th>cg00003994</th>\n",
       "      <th>cg00005847</th>\n",
       "      <th>cg00006414</th>\n",
       "      <th>cg00007981</th>\n",
       "      <th>cg00008493</th>\n",
       "      <th>cg00008713</th>\n",
       "      <th>cg00009407</th>\n",
       "      <th>cg00010193</th>\n",
       "      <th>...</th>\n",
       "      <th>cg27653134</th>\n",
       "      <th>cg27654142</th>\n",
       "      <th>cg27655855</th>\n",
       "      <th>cg27655905</th>\n",
       "      <th>cg27657249</th>\n",
       "      <th>cg27657283</th>\n",
       "      <th>cg27661264</th>\n",
       "      <th>cg27662379</th>\n",
       "      <th>cg27662877</th>\n",
       "      <th>cg27665659</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GSM1547818</th>\n",
       "      <td>0.677262</td>\n",
       "      <td>0.418543</td>\n",
       "      <td>0.03507</td>\n",
       "      <td>0.214901</td>\n",
       "      <td>0.038121</td>\n",
       "      <td>0.048343</td>\n",
       "      <td>0.984074</td>\n",
       "      <td>0.017477</td>\n",
       "      <td>0.010621</td>\n",
       "      <td>0.687097</td>\n",
       "      <td>...</td>\n",
       "      <td>0.777926</td>\n",
       "      <td>0.039206</td>\n",
       "      <td>0.818921</td>\n",
       "      <td>0.058349</td>\n",
       "      <td>0.287538</td>\n",
       "      <td>0.028253</td>\n",
       "      <td>0.245344</td>\n",
       "      <td>0.018762</td>\n",
       "      <td>0.025226</td>\n",
       "      <td>0.051665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM1547819</th>\n",
       "      <td>0.386802</td>\n",
       "      <td>0.292361</td>\n",
       "      <td>0.028753</td>\n",
       "      <td>0.316076</td>\n",
       "      <td>0.033974</td>\n",
       "      <td>0.07034</td>\n",
       "      <td>0.98456</td>\n",
       "      <td>0.016491</td>\n",
       "      <td>0.010838</td>\n",
       "      <td>0.643857</td>\n",
       "      <td>...</td>\n",
       "      <td>0.398466</td>\n",
       "      <td>0.048519</td>\n",
       "      <td>0.50977</td>\n",
       "      <td>0.059878</td>\n",
       "      <td>0.258576</td>\n",
       "      <td>0.022874</td>\n",
       "      <td>0.196758</td>\n",
       "      <td>0.015803</td>\n",
       "      <td>0.014159</td>\n",
       "      <td>0.047561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM1547820</th>\n",
       "      <td>0.523519</td>\n",
       "      <td>0.2532</td>\n",
       "      <td>0.04043</td>\n",
       "      <td>0.214164</td>\n",
       "      <td>0.043849</td>\n",
       "      <td>0.047455</td>\n",
       "      <td>0.976319</td>\n",
       "      <td>0.017809</td>\n",
       "      <td>0.010469</td>\n",
       "      <td>0.672972</td>\n",
       "      <td>...</td>\n",
       "      <td>0.767115</td>\n",
       "      <td>0.034228</td>\n",
       "      <td>0.764041</td>\n",
       "      <td>0.053957</td>\n",
       "      <td>0.263101</td>\n",
       "      <td>0.025725</td>\n",
       "      <td>0.175875</td>\n",
       "      <td>0.017959</td>\n",
       "      <td>0.027677</td>\n",
       "      <td>0.074324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM1547821</th>\n",
       "      <td>0.277912</td>\n",
       "      <td>0.836009</td>\n",
       "      <td>0.036721</td>\n",
       "      <td>0.308525</td>\n",
       "      <td>0.025322</td>\n",
       "      <td>0.090031</td>\n",
       "      <td>0.978845</td>\n",
       "      <td>0.015602</td>\n",
       "      <td>0.005646</td>\n",
       "      <td>0.671881</td>\n",
       "      <td>...</td>\n",
       "      <td>0.517407</td>\n",
       "      <td>0.043453</td>\n",
       "      <td>0.845068</td>\n",
       "      <td>0.044641</td>\n",
       "      <td>0.308361</td>\n",
       "      <td>0.021629</td>\n",
       "      <td>0.154793</td>\n",
       "      <td>0.019908</td>\n",
       "      <td>0.025395</td>\n",
       "      <td>0.041205</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM1547822</th>\n",
       "      <td>0.633324</td>\n",
       "      <td>0.435435</td>\n",
       "      <td>0.033689</td>\n",
       "      <td>0.264442</td>\n",
       "      <td>0.055146</td>\n",
       "      <td>0.037193</td>\n",
       "      <td>0.97957</td>\n",
       "      <td>0.017041</td>\n",
       "      <td>0.011099</td>\n",
       "      <td>0.640128</td>\n",
       "      <td>...</td>\n",
       "      <td>0.810477</td>\n",
       "      <td>0.045588</td>\n",
       "      <td>0.770388</td>\n",
       "      <td>0.057259</td>\n",
       "      <td>0.258144</td>\n",
       "      <td>0.021643</td>\n",
       "      <td>0.218276</td>\n",
       "      <td>0.017326</td>\n",
       "      <td>0.024316</td>\n",
       "      <td>0.077373</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM1547883</th>\n",
       "      <td>0.699218</td>\n",
       "      <td>0.466178</td>\n",
       "      <td>0.023914</td>\n",
       "      <td>0.293032</td>\n",
       "      <td>0.047161</td>\n",
       "      <td>0.14833</td>\n",
       "      <td>0.978287</td>\n",
       "      <td>0.01298</td>\n",
       "      <td>0.00742</td>\n",
       "      <td>0.660226</td>\n",
       "      <td>...</td>\n",
       "      <td>0.72812</td>\n",
       "      <td>0.058156</td>\n",
       "      <td>0.783083</td>\n",
       "      <td>0.060586</td>\n",
       "      <td>0.248578</td>\n",
       "      <td>0.016919</td>\n",
       "      <td>0.202043</td>\n",
       "      <td>0.015655</td>\n",
       "      <td>0.027396</td>\n",
       "      <td>0.041758</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM1547884</th>\n",
       "      <td>0.68797</td>\n",
       "      <td>0.491488</td>\n",
       "      <td>0.039781</td>\n",
       "      <td>0.203612</td>\n",
       "      <td>0.046104</td>\n",
       "      <td>0.085663</td>\n",
       "      <td>0.979978</td>\n",
       "      <td>0.039501</td>\n",
       "      <td>0.009376</td>\n",
       "      <td>0.638249</td>\n",
       "      <td>...</td>\n",
       "      <td>0.864897</td>\n",
       "      <td>0.057821</td>\n",
       "      <td>0.848221</td>\n",
       "      <td>0.144379</td>\n",
       "      <td>0.421445</td>\n",
       "      <td>0.04819</td>\n",
       "      <td>0.322788</td>\n",
       "      <td>0.015859</td>\n",
       "      <td>0.022922</td>\n",
       "      <td>0.145287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM1547885</th>\n",
       "      <td>0.795324</td>\n",
       "      <td>0.422477</td>\n",
       "      <td>0.03757</td>\n",
       "      <td>0.354545</td>\n",
       "      <td>0.035515</td>\n",
       "      <td>0.128453</td>\n",
       "      <td>0.986352</td>\n",
       "      <td>0.028905</td>\n",
       "      <td>0.008717</td>\n",
       "      <td>0.629348</td>\n",
       "      <td>...</td>\n",
       "      <td>0.849751</td>\n",
       "      <td>0.095023</td>\n",
       "      <td>0.762515</td>\n",
       "      <td>0.144673</td>\n",
       "      <td>0.382796</td>\n",
       "      <td>0.033938</td>\n",
       "      <td>0.281695</td>\n",
       "      <td>0.018043</td>\n",
       "      <td>0.024738</td>\n",
       "      <td>0.076221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM1547886</th>\n",
       "      <td>0.728728</td>\n",
       "      <td>0.552402</td>\n",
       "      <td>0.014772</td>\n",
       "      <td>0.197786</td>\n",
       "      <td>0.063125</td>\n",
       "      <td>0.126063</td>\n",
       "      <td>0.983295</td>\n",
       "      <td>0.018053</td>\n",
       "      <td>0.0066</td>\n",
       "      <td>0.70563</td>\n",
       "      <td>...</td>\n",
       "      <td>0.733593</td>\n",
       "      <td>0.046783</td>\n",
       "      <td>0.785088</td>\n",
       "      <td>0.05279</td>\n",
       "      <td>0.254532</td>\n",
       "      <td>0.033974</td>\n",
       "      <td>0.282958</td>\n",
       "      <td>0.01129</td>\n",
       "      <td>0.031366</td>\n",
       "      <td>0.081158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM1547887</th>\n",
       "      <td>0.729681</td>\n",
       "      <td>0.375562</td>\n",
       "      <td>0.029851</td>\n",
       "      <td>0.20372</td>\n",
       "      <td>0.062058</td>\n",
       "      <td>0.06201</td>\n",
       "      <td>0.975873</td>\n",
       "      <td>0.015436</td>\n",
       "      <td>0.010966</td>\n",
       "      <td>0.696899</td>\n",
       "      <td>...</td>\n",
       "      <td>0.711555</td>\n",
       "      <td>0.145468</td>\n",
       "      <td>0.712126</td>\n",
       "      <td>0.076845</td>\n",
       "      <td>0.236373</td>\n",
       "      <td>0.024206</td>\n",
       "      <td>0.351849</td>\n",
       "      <td>0.020035</td>\n",
       "      <td>0.03603</td>\n",
       "      <td>0.046374</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>70 rows × 27578 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "ID_REF     cg00000292 cg00002426 cg00003994 cg00005847 cg00006414 cg00007981  \\\n",
       "GSM1547818   0.677262   0.418543    0.03507   0.214901   0.038121   0.048343   \n",
       "GSM1547819   0.386802   0.292361   0.028753   0.316076   0.033974    0.07034   \n",
       "GSM1547820   0.523519     0.2532    0.04043   0.214164   0.043849   0.047455   \n",
       "GSM1547821   0.277912   0.836009   0.036721   0.308525   0.025322   0.090031   \n",
       "GSM1547822   0.633324   0.435435   0.033689   0.264442   0.055146   0.037193   \n",
       "...               ...        ...        ...        ...        ...        ...   \n",
       "GSM1547883   0.699218   0.466178   0.023914   0.293032   0.047161    0.14833   \n",
       "GSM1547884    0.68797   0.491488   0.039781   0.203612   0.046104   0.085663   \n",
       "GSM1547885   0.795324   0.422477    0.03757   0.354545   0.035515   0.128453   \n",
       "GSM1547886   0.728728   0.552402   0.014772   0.197786   0.063125   0.126063   \n",
       "GSM1547887   0.729681   0.375562   0.029851    0.20372   0.062058    0.06201   \n",
       "\n",
       "ID_REF     cg00008493 cg00008713 cg00009407 cg00010193  ... cg27653134  \\\n",
       "GSM1547818   0.984074   0.017477   0.010621   0.687097  ...   0.777926   \n",
       "GSM1547819    0.98456   0.016491   0.010838   0.643857  ...   0.398466   \n",
       "GSM1547820   0.976319   0.017809   0.010469   0.672972  ...   0.767115   \n",
       "GSM1547821   0.978845   0.015602   0.005646   0.671881  ...   0.517407   \n",
       "GSM1547822    0.97957   0.017041   0.011099   0.640128  ...   0.810477   \n",
       "...               ...        ...        ...        ...  ...        ...   \n",
       "GSM1547883   0.978287    0.01298    0.00742   0.660226  ...    0.72812   \n",
       "GSM1547884   0.979978   0.039501   0.009376   0.638249  ...   0.864897   \n",
       "GSM1547885   0.986352   0.028905   0.008717   0.629348  ...   0.849751   \n",
       "GSM1547886   0.983295   0.018053     0.0066    0.70563  ...   0.733593   \n",
       "GSM1547887   0.975873   0.015436   0.010966   0.696899  ...   0.711555   \n",
       "\n",
       "ID_REF     cg27654142 cg27655855 cg27655905 cg27657249 cg27657283 cg27661264  \\\n",
       "GSM1547818   0.039206   0.818921   0.058349   0.287538   0.028253   0.245344   \n",
       "GSM1547819   0.048519    0.50977   0.059878   0.258576   0.022874   0.196758   \n",
       "GSM1547820   0.034228   0.764041   0.053957   0.263101   0.025725   0.175875   \n",
       "GSM1547821   0.043453   0.845068   0.044641   0.308361   0.021629   0.154793   \n",
       "GSM1547822   0.045588   0.770388   0.057259   0.258144   0.021643   0.218276   \n",
       "...               ...        ...        ...        ...        ...        ...   \n",
       "GSM1547883   0.058156   0.783083   0.060586   0.248578   0.016919   0.202043   \n",
       "GSM1547884   0.057821   0.848221   0.144379   0.421445    0.04819   0.322788   \n",
       "GSM1547885   0.095023   0.762515   0.144673   0.382796   0.033938   0.281695   \n",
       "GSM1547886   0.046783   0.785088    0.05279   0.254532   0.033974   0.282958   \n",
       "GSM1547887   0.145468   0.712126   0.076845   0.236373   0.024206   0.351849   \n",
       "\n",
       "ID_REF     cg27662379 cg27662877 cg27665659  \n",
       "GSM1547818   0.018762   0.025226   0.051665  \n",
       "GSM1547819   0.015803   0.014159   0.047561  \n",
       "GSM1547820   0.017959   0.027677   0.074324  \n",
       "GSM1547821   0.019908   0.025395   0.041205  \n",
       "GSM1547822   0.017326   0.024316   0.077373  \n",
       "...               ...        ...        ...  \n",
       "GSM1547883   0.015655   0.027396   0.041758  \n",
       "GSM1547884   0.015859   0.022922   0.145287  \n",
       "GSM1547885   0.018043   0.024738   0.076221  \n",
       "GSM1547886    0.01129   0.031366   0.081158  \n",
       "GSM1547887   0.020035    0.03603   0.046374  \n",
       "\n",
       "[70 rows x 27578 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from sklearn.preprocessing import LabelEncoder\n",
    "\n",
    "encoder = LabelEncoder()\n",
    "\n",
    "test_data_matrix[\"tissue_char\"] = encoder.fit_transform(test_data_matrix[\"tissue_char\"])\n",
    "\n",
    "test_data_matrix.iloc[:,:-1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "5cbaa3d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pickle\n",
    "\n",
    "with open('datacollected.pkl', 'wb') as f:\n",
    "    pickle.dump((data_matrix_nf, test_data_matrix), f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "776cbd0c",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
